pkg/rehearse/jobs_test.go

```diff
@@ -254,38 +255,175 @@ func makeTestingProwJob(name, namespace, jobName, context string, refs *pjapi.Re
 	}
 }
 
-func TestExecuteJobs(t *testing.T) {
-	testLogger := logrus.New()
+func makeTestData() (int, string, string, *pjapi.Refs) {
 	testPrNumber := 123
 	testNamespace := "test-namespace"
-	testRepo := "testRepo"
-	testOrg := "testOrg"
 	testRefs := &pjapi.Refs{
-		Org:     testOrg,
-		Repo:    testRepo,
+		Org:     "testRepo",
+		Repo:    "testOrg",
 		BaseRef: "testBaseRef",
 		BaseSHA: "testBaseSHA",
 		Pulls:   []pjapi.Pull{{Number: testPrNumber, Author: "testAuthor", SHA: "testPrSHA"}},
 	}
+	testReleasePath := "path/to/openshift/release"
+
+	return testPrNumber, testNamespace, testReleasePath, testRefs
+}
+
+func makeSuccessfulFinishReactor(watcher watch.Interface, jobs map[string][]prowconfig.Presubmit) func(clientgo_testing.Action) (bool, watch.Interface, error) {
+	return func(clientgo_testing.Action) (bool, watch.Interface, error) {
+		watcher.Stop()
+		n := 0
+		for _, jobs := range jobs {
+			n += len(jobs)
+		}
+		ret := watch.NewFakeWithChanSize(n, true)
+		for event := range watcher.ResultChan() {
+			pj := event.Object.(*pjapi.ProwJob).DeepCopy()
+			pj.Status.State = pjapi.SuccessState
+			ret.Modify(pj)
+		}
+		return true, ret, nil
+	}
+}
+
+func TestExecuteJobsErrors(t *testing.T) {
+	testPrNumber, testNamespace, testRepoPath, testRefs := makeTestData()
+	targetRepo := "targetOrg/targetRepo"
+
+	testCases := []struct {
+		description string
+		jobs        map[string][]prowconfig.Presubmit
+		reactor     func(action clientgo_testing.Action) (handled bool, ret runtime.Object, err error)
+	}{{
+		description: "fail to Create a prowjob",
+		jobs: map[string][]prowconfig.Presubmit{targetRepo: {
+			*makeTestingPresubmit("job1", "ci/prow/job1", []string{"arg1"}),
+		}},
+		reactor: func(action clientgo_testing.Action) (bool, runtime.Object, error) {
+			return true, nil, fmt.Errorf("Fail")
+		},
+	}, {
+		description: "fail to Create one of two prowjobs",
+		jobs: map[string][]prowconfig.Presubmit{targetRepo: {
+			*makeTestingPresubmit("job1", "ci/prow/job1", []string{"arg1"}),
+			*makeTestingPresubmit("job2", "ci/prow/job2", []string{"arg2"}),
+		}},
+		reactor: func(action clientgo_testing.Action) (bool, runtime.Object, error) {
+			createAction := action.(clientgo_testing.CreateAction)
+			pj := createAction.GetObject().(*pjapi.ProwJob)
+			if pj.Spec.Job == "rehearse-123-job1" {
+				return false, nil, nil
+			}
+			return true, nil, fmt.Errorf("Fail")
+		},
+	},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			testLogger := logrus.New()
+			fakecs := fake.NewSimpleClientset()
+			fakeclient := fakecs.ProwV1().ProwJobs(testNamespace)
+			watcher, err := fakeclient.Watch(metav1.ListOptions{})
+			if err != nil {
+				t.Fatalf("Failed to setup watch: %v", err)
+			}
+			fakecs.Fake.PrependWatchReactor("prowjobs", makeSuccessfulFinishReactor(watcher, tc.jobs))
+			fakecs.PrependReactor("create", "prowjobs", tc.reactor)
```

Comment:
Could explicitly call out to `Fake.PrependReactor` or not above, to be consistent